srcpkgs/xrootd/patches/musl.patch

```diff
@@ -0,0 +1,303 @@
+diff --git a/src/Xrd/XrdConfig.hh b/src/Xrd/XrdConfig.hh
+index a06ad462f..702403d82 100644
+--- a/src/Xrd/XrdConfig.hh
++++ b/src/Xrd/XrdConfig.hh
+@@ -34,6 +34,11 @@
+ #include "Xrd/XrdProtLoad.hh"
+ #include "Xrd/XrdProtocol.hh"
+ 
++#if defined(__linux__) && !defined(__GLIBC__)
++#define __NEED_mode_t
++#include <bits/alltypes.h>
++#endif
++
+ class XrdSysError;
+ class XrdTcpMonInfo;
+ class XrdNetSecurity;
+diff --git a/src/XrdAcc/XrdAccGroups.cc b/src/XrdAcc/XrdAccGroups.cc
+index e663fdd35..14c3b19d6 100644
+--- a/src/XrdAcc/XrdAccGroups.cc
++++ b/src/XrdAcc/XrdAccGroups.cc
+@@ -46,6 +46,14 @@
+ #include "XrdAcc/XrdAccGroups.hh"
+ #include "XrdAcc/XrdAccPrivs.hh"
+ 
++#if defined(__linux__) && !defined(__GLIBC__)
++int innetgr(const char *netgroup, const char *host, const char *user,
++             const char *domain)
++{
++   return 0;
++}
++#endif
++
+ // Additionally, this routine does not support a user in more than
+ // NGROUPS_MAX groups. This is a standard unix limit defined in limits.h.
+   
+diff --git a/src/XrdApps/XrdMpxStats.cc b/src/XrdApps/XrdMpxStats.cc
+index 8268a6528..6afadf504 100644
+--- a/src/XrdApps/XrdMpxStats.cc
++++ b/src/XrdApps/XrdMpxStats.cc
+@@ -289,7 +289,7 @@ int main(int argc, char *argv[])
+    while(1)
+         {sbP = statsQ.getBuff();
+          retc = recvfrom(udpFD, sbP->Data, sizeof(sbP->Data), 0,
+-                               &sbP->From.Addr, &fromLen);
++                               &sbP->From.Addr, (socklen_t*)&fromLen);
```

Comment:
Defining `SOCKLEN_t` in `XrdSysPlatform.hh` more carefully would be better than casting `int*` to `socklen_t*`, both for upstream patch and for Void patch.